pentasharp/Controllers/BusSchedule.cs

```diff
@@ -1,21 +1,336 @@
-﻿using Microsoft.AspNetCore.Mvc;
-using System.Diagnostics;
-using WebApplication1.Models;
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using pentasharp.Data;
+using pentasharp.Models.Entities;
+using pentasharp.Models.Enums;
+using pentasharp.ViewModel.BusSchedul;
+using pentasharp.ViewModel.BusReservation;
 using WebApplication1.Filters;
+using Microsoft.AspNetCore.Authorization;
 
 namespace WebApplication1.Controllers
 {
-    [ServiceFilter(typeof(AdminOnlyFilter))]
+    [Route("api/BusSchedule")]
     public class BusScheduleController : Controller
     {
+        private readonly AppDbContext _context;
+        private readonly IMapper _mapper;
 
-        public IActionResult BusScheduleManagement()
+        public BusScheduleController(AppDbContext context, IMapper mapper)
         {
-            return View();
+            _context = context;
+            _mapper = mapper;
+        }
+
+        [ServiceFilter(typeof(AdminOnlyFilter))]
+        public IActionResult BusScheduleManagement() => View();
+
+        [ServiceFilter(typeof(AdminOnlyFilter))]
+        [HttpPost("AddRoute")]
+        public IActionResult AddRoute([FromBody] AddRouteViewModel model, int hours, int minutes)
+        {
+            if (!ModelState.IsValid || string.IsNullOrWhiteSpace(model.FromLocation) || string.IsNullOrWhiteSpace(model.ToLocation))
```

Comment:
use linq here and in the other if check as well, seperate them into variables and only call the variable in the if check